resotocore/tools/render_dot.py

```diff
@@ -0,0 +1,212 @@
+from typing import Generator, List, Dict, Iterator, Sequence, Mapping, Union, Any, Optional
+import json
+from enum import Enum
+import os
+import requests
+import zipfile
+import argparse
+from resotocore.util import (
+    value_in_path,
+    value_in_path_get,
+    count_iterator,
+)
+from resotocore.model.resolve_in_graph import NodePath
+from resotocore.util import uuid_str, utc
+from collections import defaultdict
+import re
+from dataclasses import dataclass
+from posthog import Client
+
+JsonElement = Union[str, int, float, bool, None, Mapping[str, Any], Sequence[Any]]
+
+class ResourceKind(Enum):
+    INSTANCE = 1
+    VOLUME = 2
+    IMAGE = 3
+    FIREWALL = 4
+    K8S_CLUSER = 5
+    NETWORK = 6
+    LOAD_BALANCER = 7
+    CLOUD = 8
+
+kind_colors = {
+    ResourceKind.INSTANCE: "8",
+    ResourceKind.VOLUME: "4",
+    ResourceKind.IMAGE: "7",
+    ResourceKind.FIREWALL: "6",
+    ResourceKind.K8S_CLUSER: "5",
+    ResourceKind.NETWORK: "10",
+    ResourceKind.LOAD_BALANCER: "9",
+    ResourceKind.CLOUD: "1",
+}
+
+@dataclass
+class ResourceDescription:
+    uid: str
+    name: str
+    id: str
+    kind: ResourceKind
+    kind_name: str
+
+
+do_kinds = {
+    "droplet": ResourceKind.INSTANCE,
+    "volume": ResourceKind.VOLUME,
+    "image": ResourceKind.IMAGE,
+    "firewall": ResourceKind.FIREWALL,
+    "kubernetes_cluster": ResourceKind.K8S_CLUSER,
+    "network": ResourceKind.NETWORK,
+    "load_balancer": ResourceKind.LOAD_BALANCER,
+}
+
+
+def parse_kind(kind: str) -> Optional[ResourceKind]:
```

Comment:
It would be straightforward to fetch the model from the core.
This way you could use the base kinds like instance, volume etc. as fallback and specific icons if available (e.g. aws).

Reply:
We decided to postpone this for now.